packages/features/bookings/Booker/components/Header.tsx

```diff
@@ -0,0 +1,111 @@
+import { ChevronLeft, ChevronRight } from "lucide-react";
+import { useCallback } from "react";
+import { shallow } from "zustand/shallow";
+
+import dayjs from "@calcom/dayjs";
+import { useLocale } from "@calcom/lib/hooks/useLocale";
+import { Button, ButtonGroup, ToggleGroup } from "@calcom/ui";
+import { Calendar, Columns, Grid } from "@calcom/ui/components/icon";
+
+import { TimeFormatToggle } from "../../components/TimeFormatToggle";
+import { useBookerStore } from "../store";
+
+export function Header({ extraDays, isMobile }: { extraDays: number; isMobile: boolean }) {
+  const [layout, setLayout] = useBookerStore((state) => [state.layout, state.setLayout], shallow);
+  const selectedDateString = useBookerStore((state) => state.selectedDate);
+  const addToSelectedDate = useBookerStore((state) => state.addToSelectedDate);
+  const isSmallCalendar = layout === "small_calendar";
+  const selectedDate = dayjs(selectedDateString);
+
+  const onLayoutToggle = useCallback((newLayout: BookerLayout) => setLayout(newLayout), [setLayout]);
+
+  if (isMobile) return null;
+
+  // In month view we only show the layout toggle.
+  if (isSmallCalendar) {
+    return (
+      <div className="fixed top-3 right-3 z-10">
+        <LayoutToggle onLayoutToggle={onLayoutToggle} layout={layout} />
+      </div>
+    );
+  }
+
+  return (
+    <div className="border-subtle relative z-10 flex border-l border-b p-4">
+      <div className="flex items-center gap-3">
+        <h3 className="text-base font-semibold leading-4">
+          {selectedDate.format("MMM D")}-{selectedDate.add(extraDays, "days").format("D")},{" "}
+          <span className="text-subtle">{selectedDate.format("YYYY")}</span>
+        </h3>
+        <ButtonGroup>
+          <Button
+            variant="icon"
+            color="minimal"
+            StartIcon={ChevronLeft}
+            aria-label="Previous Day"
+            onClick={() => addToSelectedDate(-extraDays - 1)}
+          />
+          <Button
+            variant="icon"
+            color="minimal"
+            StartIcon={ChevronRight}
+            aria-label="Next Day"
+            onClick={() => addToSelectedDate(extraDays + 1)}
+          />
+        </ButtonGroup>
+      </div>
+      <div className="ml-auto flex gap-3">
+        <TimeFormatToggle />
+        <div className="fixed top-4 right-4">
+          <LayoutToggle onLayoutToggle={onLayoutToggle} layout={layout} />
+        </div>
+        {/*
+          This second layout toggle is hidden, but needed to reserve the correct spot in the DIV
```

Comment:
Couldn't come up with a better solution. Don't think this is too bad, but also not 100% happy with it. For now it works well I think, maybe later we can see if we can come up with a better solution.

Reply:
I was actually curious how this was gonna be tackled ... easier than i had it in my head :P

Reply:
Hahaha. I tried a lot with framer motion too, but you always saw the jump. So instead of spending days on an animation I think this for now was the best solution, haha.